src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -260,7 +271,9 @@ static ResourceDefinition getKeyManagerDefinition() {
                         throw new StartException(e);
                     }
 
-                    return keyManagerFactory.getKeyManagers();
+                    KeyManager[] keyManagers = keyManagerFactory.getKeyManagers();
+                    if (alias != null) wrapKeyManagersByAliasSelector(keyManagers, alias);
```

Comment:
Instead of wrapping the KeyManager it is the KeyStore that should be wrapped using the filtering key store implementation from Elytron and the alias filter defined on this resource.